### PR TITLE
Adds tests for regex edge cases

### PR DESCRIPTION
**Pull request by alexkalderimis**
When using regular expressions, $ and _ will match differently if the
user uses \W or \w - these tests test for these edge cases.

### CODE REVIEW

File: bob/bob.t [hunk 1]
```diff
@@ -28,6 +28,10 @@ my @cases = map {
     ['ZOMG THE %^*@#$(*^ ZOMBIES ARE COMING!!11!!1!',  'Whoa, chill out!',   'shouting with special characters'],
     ["ÜMLÄÜTS!",                                       'Whoa, chill out!',   'shouting with umlauts'],
     ["\xdcML\xc4\xdcTS!",                              'Whoa, chill out!',   'shouting with umlauts'],
+    ['_A',                                             'Whoa, chill out!',   'Dollar sign shout - before'],
```
Comment (darksuji) [inline bob/bob.t]: Whoops, looks like you flipped the names on these.

**Comment by kytrinyx** [comment]
Thanks!
